src/data/html/highlight-hover.js

```diff
@@ -1,32 +1,35 @@
-// Copyright 2002-2010, Simon Marlow.  All rights reserved.
-// https://github.com/haskell/haddock/blob/ghc-8.8/LICENSE
-// Slightly modified by Tesla Ice Zhang
+// Copyright 2023, Andreas Abel.
+// Falls under the Agda license at https://github.com/agda/agda/blob/master/LICENSE
 
-var highlight = function (on) {
-  return function () {
-    var links = document.getElementsByTagName('a');
-    for (var i = 0; i < links.length; i++) {
-      var that = links[i];
+// When we hover over an Agda identifier, we highlight all occurrences of this identifier on the page.
+// To this end, we create a map from identifier to all of its occurrences in the beginning.
 
-      if (this.href != that.href) {
-        continue;
-      }
+// A dictionary from hrefs to 'a'-elements that have this href.
+var dict = new Map();
 
-      if (on) {
-        that.classList.add("hover-highlight");
-      } else {
-        that.classList.remove("hover-highlight");
-      }
-    }
+window.onload = function () {
+
+  // Get all 'a' tags with an 'href' attribute.
+  // We call those "objects".
+  const objs  = [...document.getElementsByTagName('a')].filter((obj) => obj.hasAttribute('href'));
```

Comment:
```suggestion
  const objs  = document.getSelectorAll('a[href]');
```

[attribute selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors). since all you do over `objs` is for-of iteration you don't need to explode it into an array, either

Reply:
Thanks, I am an absolute JS beginner.

Reply:
Almost 😅 https://github.com/agda/agda/pull/6948